### PR TITLE
Sales

### PR DESCRIPTION
**Pull request by sergeyq**

### CODE REVIEW

File: src/components/helpers/fields/saleItemField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  field: ISaleItem;
+};
+
+export default class SaleItemField extends React.Component<IProps, any> {
+
+  public render(): JSX.Element {
+    const { field } = this.props;
+
+    return (
+      <div className='sales-row' id={field.id}>
```
Comment (DmitryLeviossa) [inline src/components/helpers/fields/saleItemField/index.tsx]: Why do you need id?

Comment (DmitryLeviossa) [reply src/components/helpers/fields/saleItemField/index.tsx]: Send id with bind(this, field.id)

File: src/components/management/companyUsers/UserFormModal/index.tsx [hunk 2]
```diff
@@ -96,9 +100,10 @@ export default class UserFormModal extends React.Component<IProps, any> {
     return Object.assign({}, ...arrFields);
   }
 
-  private validate({ isValidateUserName = true, isValidateEmail = true,
-     isValidateRoleId = true, isValidateDepartmentId = true, isValidateFirstName = true, isValidateLastName = true}) {
+  private validate({ isValidateUserName = true, isValidateEmail = true, isValidateRoleId = true,
+    isValidateDepartmentId = true, isValidateFirstName = true, isValidateLastName = true, isValidateSales = false }) {
```
Comment (DmitryLeviossa) [inline src/components/management/companyUsers/UserFormModal/index.tsx]: Don't like that implementation, no need to send so many attributes

File: src/components/management/companyUsers/UserFormModal/index.tsx [hunk 3]
```diff
@@ -125,6 +130,13 @@ export default class UserFormModal extends React.Component<IProps, any> {
       role_id.note = '';
     }
 
+    if (isValidateSales) {
+      sales.forEach((sale) => {
```
Comment (DmitryLeviossa) [inline src/components/management/companyUsers/UserFormModal/index.tsx]: User interfaces

File: src/components/management/companyUsers/UserFormModal/index.tsx [hunk 4]
```diff
@@ -179,9 +191,22 @@ export default class UserFormModal extends React.Component<IProps, any> {
       }
     }
 
+    if (isValidateSales && sales.some((sale) => (!sale.name || !sale.description))) {
+      sales.forEach((sale) => {
```
Comment (DmitryLeviossa) [inline src/components/management/companyUsers/UserFormModal/index.tsx]: Use interfaces

File: src/components/management/companyUsers/UserFormModal/index.tsx [hunk 5]
```diff
@@ -248,17 +277,21 @@ export default class UserFormModal extends React.Component<IProps, any> {
...
-    // }
+    }
   }
 
   @bind
   private async editUser(e): Promise<any> {
     const { username, email, role_id, department_id, first_name, last_name } = this.state.form;
-    const { selectedUser } = this.state;
+    const { selectedUser, sales } = this.state;
+    const newSales = sales.filter( (i) => i.isNew).map(({ name, description }) => ({name, description }));
```
Comment (DmitryLeviossa) [inline src/components/management/companyUsers/UserFormModal/index.tsx]: interfaces

File: src/components/management/companyUsers/UserFormModal/index.tsx [hunk 6]
```diff
@@ -295,36 +330,51 @@ export default class UserFormModal extends React.Component<IProps, any> {
...
-        <CustomSelect field={this.state.form.role_id} type='text' name='role_id'
-                      selectFields={this.props.roles} onChangeField={this.handleInput} />
-        <CustomSelect field={this.state.form.department_id} type='text' name='department_id'
-                      selectFields={this.props.departments} onChangeField={this.handleInput}/>
-        {/*{isEdit ? <CustomDatePicker field={this.state.form.birthday} type='text' name='birthday'*/}
-                                    {/*onChangeDateField={this.handleDateInput}/> : ''}*/}
-        
+      <div>
+        <div className='modal-inner'>
+          {isEdit && <CustomText field={this.state.form.username} type='text' name='username'
```
Comment (DmitryLeviossa) [inline src/components/management/companyUsers/UserFormModal/index.tsx]: don't user ternary operator so much, create separate functions

File: src/components/management/companyUsers/UserFormModal/index.tsx [hunk 7]
```diff
@@ -295,36 +330,51 @@ export default class UserFormModal extends React.Component<IProps, any> {
...
+                          onAddSaleItem={this.addSaleItem}
+                          onDel={this.handleSalesDel} />}
       </div>
     );
   }
+
+  private addSaleItem = () => {
+    this.setState((prev) => ({
+      sales: [...prev.sales, ...[
+        { id: uniqueId(), name: '', description: '', error: '', note: '', isNew: true },
```
Comment (DmitryLeviossa) [inline src/components/management/companyUsers/UserFormModal/index.tsx]: like it!

File: src/components/management/companyUsers/UserFormModal/index.tsx (hunk 2 above)
Comment (sergeyq) [reply src/components/management/companyUsers/UserFormModal/index.tsx]: I can suggest using the redux form for better control validation fields.

File: src/components/management/companyUsers/UserFormModal/index.tsx [hunk 8]
```diff
@@ -96,92 +100,42 @@ export default class UserFormModal extends React.Component<IProps, any> {
...
-    if (isValidateEmail) {
-      email.error = false;
-      email.note = '';
-    }
+  private validate({
+   isAllRequired = false,
+   isValidateSales = false,
+  }) {
+    const { sales } = this.state;
+    const { username, email, department_id, role_id, first_name, last_name } = this.setValidateFields(isAllRequired);
```
Comment (sergeyq) [inline src/components/management/companyUsers/UserFormModal/index.tsx]: I hope I understood correctly what was required

File: src/components/applications/index.tsx [hunk 9]
```diff
@@ -72,7 +72,7 @@ export default class Applications extends React.Component<IProps, any> {
         {this._renderFilters()}
         <div className='applications-head'>
           <div className='applications-header'>Applications</div>
-          <PlusButton onClick={this.openModal}/>
+          <PlusButton class='plus-button' handleClick={this.openModal}/>
```
Comment (DmitryLeviossa) [inline src/components/applications/index.tsx]: className, change also props in component

**Pull request merged**